shared/web_features.go

```diff
@@ -0,0 +1,109 @@
+// Copyright 2024 The WPT Dashboard Project. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package shared
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+)
+
+// WebFeaturesData is the public data type that represents the data parsed from
+// a given manifest file
+type WebFeaturesData map[string]map[string]interface{}
+
+// ErrUnknownWebFeaturesManifestVersion indicates that the parser does not know how to parse
+// this version of the web features file.
+var ErrUnknownWebFeaturesManifestVersion = errors.New("unknown web features manifest version")
+
+// ErrBadWebFeaturesManifestJSON indicates that there was an error parsing the given
+// v1 manifest file.
+var ErrBadWebFeaturesManifestJSON = errors.New("invalid json when reading web features manifest")
+
+// ErrUnexpectedWebFeaturesManifestV1Format indicates that there was an error parsing the given
+// v1 manifest file.
+var ErrUnexpectedWebFeaturesManifestV1Format = errors.New("unexpected web features manifest v1 format")
+
+// TestMatchesWithWebFeature performs two checks.
+// If the given test path is present in the data. If not, return false
+// If it is present, check if the given web feature applies to that test.
+func (d WebFeaturesData) TestMatchesWithWebFeature(test, webFeature string) bool {
+	if len(d) == 0 {
+		return false
+	}
+	if webFeatures, ok := d[test]; ok {
+		_, found := webFeatures[strings.ToLower(webFeature)]
+
```

Comment:
```suggestion
```

Reply:
I used the golangci-lint tool (currently it only applies to the api folder now, eventually it should apply to other folders. I decided to use it on these files to help with the migration to it whenever it happens). It comes up with a [nlreturn](https://github.com/ssgreg/nlreturn) lint error. When the code block is above a certain size, it requires a new line before the return.